### PR TITLE
imapK implementation for RabbitClient

### PR DESCRIPTION
**Pull request by jbwheatley**

### CODE REVIEW

File: core/src/main/scala/dev/profunktor/fs2rabbit/interpreter/RabbitClient.scala [hunk 1]
```diff
@@ -79,17 +80,31 @@ object RabbitClient {
...
+        client.connection.mapK(fK),
+        client.consume.imapK(fK)(gK),
+        client.publish.imapK(fK)(gK),
+        client.binding.mapK(fK),
+        client.declaration.mapK(fK),
+        client.deletion.mapK(fK),
+        client.consumingProgram.imapK(fK)(gK),
+        client.publishingProgram.imapK(fK)(gK)
+      )
+  }
```
Comment (jbwheatley) [inline core/src/main/scala/dev/profunktor/fs2rabbit/interpreter/RabbitClient.scala:96]: changes allow for this

File: core/src/main/scala/dev/profunktor/fs2rabbit/program/ConsumingProgram.scala [hunk 2]
```diff
@@ -71,10 +71,10 @@ case class WrapperConsumingProgram[F[_]: Sync] private (
         case (tag, _) =>
           consume.basicCancel(channel, tag)
       }
-      .flatMap {
+      .evalMap {
         case (_, queue) =>
-          queue.dequeue.rethrow
-            .evalMap(env => decoder(env).map(a => env.copy(payload = a)))
+          queue.take.rethrow
```
Comment (jbwheatley) [inline core/src/main/scala/dev/profunktor/fs2rabbit/program/ConsumingProgram.scala:76]: this needs fixing.. we want to dequeue the entire queue, not just the head

Comment (jbwheatley) [reply core/src/main/scala/dev/profunktor/fs2rabbit/program/ConsumingProgram.scala:76]: waiting for `Stream.fromQueueUnterminated` in next fs2 milestone

**Comment by jbwheatley** [comment]
this could also be done by adding `trait RabbitClient[F[_]]` and changing `class RabbitClient[F[_]]` -> `class RabbitClientImpl[F[_]] extends RabbitClient[F]`

File: core/src/main/scala/dev/profunktor/fs2rabbit/algebra/AMQPInternals.scala [hunk 3]
```diff
@@ -17,6 +17,9 @@
 package dev.profunktor.fs2rabbit.algebra
 
 import cats.effect.std.Queue
+import cats.~>
 import dev.profunktor.fs2rabbit.model.AmqpEnvelope
 
-case class AMQPInternals[F[_]](queue: Option[Queue[F, Either[Throwable, AmqpEnvelope[Array[Byte]]]]])
+case class AMQPInternals[F[_]](queue: Option[Queue[F, Either[Throwable, AmqpEnvelope[Array[Byte]]]]]) {
+  def mapK[G[_]](fK: F ~> G): AMQPInternals[G] = AMQPInternals(queue.map(_.mapK(fK)))
```
Comment (agustafson) [inline core/src/main/scala/dev/profunktor/fs2rabbit/algebra/AMQPInternals.scala:24]: Can this be auto-derived? 🤔

Comment (jbwheatley) [reply core/src/main/scala/dev/profunktor/fs2rabbit/algebra/AMQPInternals.scala:24]: don't think so, i think auto-derive only works for things of the shape `def foo: F[X]`

**Pull request closed without merging**